atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/KvsLockWatchingService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import java.util.Set;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.common.annotation.Idempotent;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.watch.LockWatchState;
+
+public interface KvsLockWatchingService {
```

Comment:
Non-actionable: I gather that this is named this way because this class is interested in Atlas table level concerns (like `TableReference` or rows) while the more general service works in terms of `LockDescriptor`. Not entirely happy with the name, but don't really have a better idea right now.

Reply:
Yeah, it used to be called `TransactionLockWatchingService`, so it's improving...